src/org/opendatakit/briefcase/model/form/FileSystemFormMetadataAdapter.java

```diff
@@ -0,0 +1,137 @@
+package org.opendatakit.briefcase.model.form;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.util.stream.Collectors.toMap;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.delete;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.walk;
+import static org.opendatakit.briefcase.reused.UncheckedFiles.write;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import org.opendatakit.briefcase.export.XmlElement;
+import org.opendatakit.briefcase.model.FormKey;
+import org.opendatakit.briefcase.reused.BriefcaseException;
+import org.opendatakit.briefcase.reused.LegacyPrefs;
+
+public class FileSystemFormMetadataAdapter implements FormMetadataPort {
+  private static final ObjectMapper MAPPER = new ObjectMapper().findAndRegisterModules();
+  private final Map<FormKey, FormMetadata> store = new ConcurrentHashMap<>();
+
+  public FileSystemFormMetadataAdapter() {
+  }
+
+  public static FormMetadataPort at(Path storageRoot) {
+    return new FileSystemFormMetadataAdapter().syncWithFilesAt(storageRoot);
+  }
+
+  private static JsonNode uncheckedReadTree(Path jsonFile) {
+    try {
+      return MAPPER.readTree(jsonFile.toFile());
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public FormMetadataPort syncWithFilesAt(Path storageRoot) {
+    flush();
+    store.putAll(walk(storageRoot.resolve("forms"))
+
+        // select XML files that are not submissions
+        .filter(path -> !path.getFileName().toString().equals("submission.xml")
+            && path.getFileName().toString().endsWith(".xml"))
```

Comment:
`path.toString().endsWith` also works. I suspect you know this and wrote it the way you did deliberately.

Reply:
Yeah, I want to give an explicit hint about our focus on the filename.